src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,300 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+# the License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+from abc import ABC, abstractmethod
+
+import boto3
+import yaml
+from botocore.exceptions import ClientError
+from slurm_plugin.slurm_resources import EC2Instance
```

Comment:
optional: it would be nice to drop the dependency of fleet manager on something which is defined in slurm_plugin. I'd rather expect the other way around. Would it make sense to move EC2Instance out of the slurm modules?

Reply:
Done, moved to instance_manager.py module.